src/game/data.py

```diff
@@ -40,3 +40,79 @@ class GameTurn:
     def __init__(self):
         self.player_one_active = True
         self.selected_piece = 0
+
+
+class GameState:
+    """Definition of all data of the game at an instant:
+    - grid
+    - remaining_pieces
+    - game_turn"""
+
+    def __init__(self, parameter=""):
+        self.grid = GameState.init_grid()
+        self.remaining_pieces = GameState.init_remaining_pieces()
+        self.game_turn = GameState.init_game_turn()
+        self.message = ""
+        if type(parameter) is dict:
+            self.load_state(parameter)
+
+    def load_state(self, parameter):
+        try:
+            for key, val in parameter["grid"].items():
+                self.place_piece(key, val)
+
+            self.game_turn.player_one_active = parameter["turn"]["player"] == 1
+            self.game_turn.selected_piece = parameter["turn"]["selected"]
+        except:
```

Comment:
Catchall `except` isn't recommended at all, since it can hide type errors and other real errors